### PR TITLE
Support session cookies.

### PR DESCRIPTION
**Pull request by connelly38**
If set in response headers, allow setting a cookie named "session" to enable session persistence in load balancer routing.

### CODE REVIEW

File: driver/src/main/java/oracle/nosql/driver/http/Client.java [hunk 1]
```diff
@@ -546,6 +553,9 @@ public Result execute(Request kvRequest) {
                 headers.add(HttpHeaderNames.HOST, host)
                     .add(REQUEST_ID_HEADER, requestId)
                     .setInt(CONTENT_LENGTH, contentLength);
+                if (sessionCookie != null && sessionCookie.length() > 0) {
+                    headers.add(COOKIE, SESSION_COOKIE_FIELD + sessionCookie);
```
Comment (gmfeinberg) [inline driver/src/main/java/oracle/nosql/driver/http/Client.java:557]: why not include the "session=" in the value of the cookie?

File: driver/src/main/java/oracle/nosql/driver/http/Client.java [hunk 2]
```diff
@@ -1037,6 +1051,36 @@ private void processNotOKResponse(HttpResponseStatus status,
                                  ", reason = " + status.reasonPhrase());
     }
 
+    /* set session cookie, if set in response headers */
+    private void setSessionCookie(HttpHeaders headers) {
+        if (headers == null) {
+            return;
+        }
+        List<String> values = headers.getAll("Set-Cookie");
```
Comment (gmfeinberg) [inline driver/src/main/java/oracle/nosql/driver/http/Client.java:1059]: I think you can get by with the simple "get" for this which simplifies the logic here

File: driver/src/main/java/oracle/nosql/driver/http/Client.java (hunk 1 above)
Comment (connelly38) [reply driver/src/main/java/oracle/nosql/driver/http/Client.java:557]: ok I can do that. will save an alloc/create per request.

File: driver/src/main/java/oracle/nosql/driver/http/Client.java (hunk 2 above)
Comment (connelly38) [reply driver/src/main/java/oracle/nosql/driver/http/Client.java:1059]: If there's more than one Set-Cookie header (not likely, but _possible_), then get() only gets the first one. Hence this code, to be sure.

Comment (gmfeinberg) [reply driver/src/main/java/oracle/nosql/driver/http/Client.java:1059]: I don't think that multiple Set-Cookie headers is legitimate, at least in our path. getAll() creates a new List for results and get() does not (from my reading of Netty)

Comment (connelly38) [reply driver/src/main/java/oracle/nosql/driver/http/Client.java:1059]: OK, I'll change it and add a comment. If the LB ever does try to set two cookies then this won't work.

**Pull request closed without merging**